source/extensions/filters/udp/udp_proxy/config.h

```diff
@@ -14,6 +14,43 @@ namespace UdpProxy {
 using TunnelingConfig =
     envoy::extensions::filters::udp::udp_proxy::v3::UdpProxyConfig::UdpTunnelingConfig;
 
+/**
+ * Base class for both tunnel response headers and trailers.
+ */
+class TunnelResponseHeadersOrTrailers : public StreamInfo::FilterState::Object {
+public:
+  ProtobufTypes::MessagePtr serializeAsProto() const override;
+  virtual const Http::HeaderMap& value() const PURE;
+};
+
+/**
+ * Response headers for the tunneling connections.
+ */
+class TunnelResponseHeaders : public TunnelResponseHeadersOrTrailers {
```

Comment:
@kyessenov this is basically replicating what you did in https://github.com/envoyproxy/envoy/pull/23118 but for UDP. Looks fine to me, do you have any opinion on whether it is worth refactoring and sharing code across the implementations or just going with what is in this PR?

Reply:
ping @kyessenov

Reply:
Sorry, was OoO. I'm fine if we postpone code consolidation later. These are tiny classes and fairly easy to refactor.

Reply:
Thanks @kyessenov .
@htuch Do you have other comments on the PR, please?